File: Java/factory/src/challenge/PacificCalendar.java
package challenge;

import java.util.List;

public class PacificCalendar extends Calendar{

    public PacificCalendar(ZoneFactory zoneFactory){
        zone = zoneFactory.createZone("US/Pacific");
    }

    @Override
    public void createCalendar(List<String> appointments) {
        System.out.println("Making the calendar");
    }
}

File: C++/factory/challenge/PacificCalendar.cpp
//
// Created by Chans on 2020/2/6.
//

#include "PacificCalendar.h"

PacificCalendar::PacificCalendar(ZoneFactory *zoneFactory) {
    factory = zoneFactory;
    zone = zoneFactory->CreateZone();
}

void PacificCalendar::createCalendar(list<string> appointments) {
    cout <<  "Making the calendar" << endl;
};
